src/compiler/parse/state/tag.ts

```diff
@@ -387,12 +387,22 @@ function read_attribute(parser: Parser, unique_names: Set<string>) {
 			}, start);
 		}
 
-		if (value[0]) {
-			if ((value as any[]).length > 1 || value[0].type === 'Text') {
-				parser.error({
-					code: 'invalid-directive-value',
-					message: 'Directive value must be a JavaScript expression enclosed in curly braces'
-				}, value[0].start);
+		const first_value = value[0];
+		let expression = null;
+		let text = null;
+
+		if (first_value) {
+			if ((value as any[]).length > 1 || first_value.type === 'Text') {
+				if (type === 'Style') {
+					text = first_value.data;
+				} else {
```

Comment:
I'm struggling to remember the rationale for that restriction. It's very possible that it's a hangover from Svelte 2 (or 1!) when the syntax was different. 

I think it probably makes sense to accept string values, just as we do in attributes and props. `"red"` and `"translateX({translate_x})"` are both good examples of where this would be useful, but I can even see this being useful in other contexts:

```svelte
<div use:portal="#my-element">...</div>
```